src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Handlers/RetryHandler.cs

```diff
@@ -74,17 +74,22 @@ public class RetryHandler : DelegatingHandler
                 }
                 catch (Exception ex)
                 {
-                    if (ex.IsFatal() || attempt == RetryHandler.MaxAttempts)
+                    if (ex.IsFatal() || attempt == RetryHandler.MaxAttempts - 1)
```

Comment:
And combining with my response above if we want to throw if we have reached max attempt, initialize attempt outside of the for loop and if  attempt > RetryHandler.MaxAttempts then throw outside of the for loop

---
In reply to: [66560166](https://github.com/Azure/azure-powershell/pull/2412#discussion_r66560166) [](ancestors = 66560166,66539492)